friends/src/Components/Form.js

```diff
@@ -0,0 +1,53 @@
+import React, { Component } from "react";
+import { Link } from "react-router-dom";
+
+class Form extends Component {
+  submitHandler = event => {
+    event.preventDefault();
+    this.props.addFriend(this.props.state);
+    this.props.history.push("/");
+  };
+
+  render() {
+    return (
+      <>
+        <form className="friends-list" onSubmit={this.submitHandler}>
+          <input
+            autoComplete="off"
+            className="input-section"
+            type="text"
+            name="name"
+            placeholder="Name"
+            value={this.props.state.name}
+            onChange={this.props.handleChange}
+          />
+          <input
+            autoComplete="off"
+            className="input-section"
+            type="number"
+            name="age"
+            placeholder="Age"
+            value={this.props.state.age}
+            onChange={this.props.handleChange}
+          />
+          <input
+            autoComplete="off"
+            className="input-section"
+            type="text"
+            name="email"
+            placeholder="Email"
+            value={this.props.state.email}
+            onChange={this.props.handleChange}
+          />
```

Comment:
Your inputs look very similar. Why not make a helper function for them? For example:
```javascript
render() {
  const makeInput = (name, type='text') => (
    <input
      autoComplete="off"
      className="input-section"
      type={type}
      name={name}
      placeholder={name[0].toUpperCase() + name.slice(1)}
      value={this.props.state[name]}
      onChange={this.props.handleChange}
    />
  );
  return (
    <>
      <form className="friends-list" onSubmit={this.submitHandler}>
        { makeInput('name') }
        { makeInput('age', 'number') }
        { makeInput('email') }
        // ...
```

Reply:
The same applies to your Edit page (maybe you could make an actual component and then specify some of the props within each to make it similar to this.